java/core/src/test/org/apache/orc/impl/TestOrcLargeStripe.java

```diff
@@ -214,4 +215,77 @@ public void testStringDirectGreaterThan2GB() throws IOException {
       fs.delete(testFilePath, false);
     }
   }
+
+  @Test
+  public void testRowBatchSizeWhenStringDirectGreaterThan2GB() throws IOException {
+    final Runtime rt = Runtime.getRuntime();
+    assumeTrue(rt.maxMemory() > 4_000_000_000L);
```

Comment:
No, increasing the memory requirements is harmful because we will miss the memory leakage issues in the future development. That's the point why we requires `2g` in our tests. Can you revise the test case instead?

Reply:
Make sense.
Now the trigger condition is that a relatively large string object is required. I tried the size as much as possible in this test case, but 2g is still not enough.
I now use the mock method to test this exception.